skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,197 @@
+
+    Parameters
+    ----------
+    image : ndarray
+        The image to be filtered.
+    kernel_shape: ndarray or scalar, optional
+        The length of the non-intensity vertices of the ellipsoid. If
+        ``kernel_shape`` is a ndarray, it must have the same dimensions as
+        ``image``. If ``kernel_shape`` is a scalar it will be extended to
+        have the same dimension via
+        ``kernel_shape = kernel_shape * np.ones_like(image.shape)``.
+        All elements must be greater than 0.
+    intensity_vertex : scalar, optional
+        The length of the intensity vertex of the ellipsoid. Must be greater
+        than 0.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+    num_threads: int, optional
+        The maximum number of threads to use. If ``None`` use the OpenMP
+        default value; typically equal to the maximum number of virtual cores.
+        Note: This is an upper limit to the number of threads. The exact number
+        is determined by the system's OpenMP library.
+
+    Returns
+    -------
+    background : ndarray
+        The estimated background of the image.
+
+    Notes
+    -----
+    - For the pixel that has its background intensity estimated (without loss
+      of generality at ``(0,0)``) the rolling ellipsoid method places an
+      ellipsoid under it and raises the ellipsoid until its surface touches the
+      image umbra at ``pos=(y,x)``. The background intensity is then estimated
+      using the image intensity at that position (``image[y, x]``) plus the
+      difference of ``intensity_vertex`` and the surface of the ellipsoid at
+      ``pos``. The surface intensity of the ellipsoid is computed using the
+      canonical ellipsis equation
+      .. code-block:: python
+
+        | semi_spatial = kernel_shape / 2
+        | semi_vertex = intensity_vertex / 2
+        | np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assumes that dark pixels correspond to the background. If
+      you have a bright background, invert the image before passing it to the
+      function, e.g., using `utils.invert`.
+    - This algorithm is sensitive to noise (in particular salt-and-pepper
+      noise). If this is a problem in your image, you can apply mild
+      gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ellipsoid
+    >>> image = data.coins()
+    >>> background = rolling_ellipsoid(data.coins())
+    >>> filtered_image = image - background
+    """
+
+    if num_threads is None:
+        num_threads = 0
```

Comment:
Why not `num_threads=0` in the signature?

Reply:
Because `None` is consistent with pre-existing multi-threaded code, but `0` seems required to be compatible with cython/openMP.

Incidentally, this seems to be the first openMP multi-threaded function that allows control over the number of threads in the library; otherwise I would have matched the style of those functions.